qa-tests-backend/src/main/groovy/util/CollectorUtil.groovy

```diff
@@ -0,0 +1,109 @@
+package util
+
+import static util.Helpers.withRetry
+
+import groovy.util.logging.Slf4j
+import groovy.transform.CompileStatic
+
+import orchestratormanager.Kubernetes
+
+import com.google.protobuf.util.JsonFormat
+
+import sensor.Collector
+
+@Slf4j
+@CompileStatic
+class CollectorUtil {
+    static final String RUNTIME_CONFIG_MAP_NAME = "collector-config"
+    static final String RUNTIME_CONFIG_MAP_KEY = "runtime_config.yaml"
+
+    static final String ENABLED_VALUE = "ENABLED"
+    static final String DISABLED_VALUE = "DISABLED"
+
+    static parseJsonToProtobuf(String json) {
+        Collector.CollectorConfig.Builder builder = Collector.CollectorConfig.newBuilder()
+        JsonFormat.parser().merge(json, builder)
+        return builder.build()
+    }
+
+    static introspectionQuery(String collectorAddress, String endpoint) {
+        String uri = "http://${collectorAddress}${endpoint}"
+        URL url = new URL(uri)
+        HttpURLConnection connection = null
+
+        try {
+            connection = (HttpURLConnection) url.openConnection()
+
+            // this might be unneeded?
+            connection.setRequestMethod("GET")
+            connection.connect()
+
+            if (connection.responseCode != HttpURLConnection.HTTP_OK) {
+                throw new RuntimeException("IntrospectionQuery failed with ${connection.responseMessage}")
+            }
+            def jsonResponse = connection.inputStream.text
+            return parseJsonToProtobuf(jsonResponse)
+        } catch (Exception e) {
+            throw new RuntimeException("Error making request: ${e.message}", e)
```

Comment:
Why do you rethrow this exception? It will be caught again in `withRetry`

Reply:
I have removed this try catch. I initially put it there before using withRetry.